src/test/java/spoon/test/compilationunit/GetBinaryFilesTest.java

```diff
@@ -16,36 +16,41 @@
  */
 package spoon.test.compilationunit;
 
-import org.junit.Rule;
-import org.junit.Test;
-import org.junit.rules.TemporaryFolder;
-import spoon.Launcher;
-import spoon.SpoonModelBuilder;
-import spoon.reflect.cu.CompilationUnit;
 
 import java.io.File;
 import java.io.IOException;
+import java.nio.file.Path;
 import java.util.List;
 import java.util.Map;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNotNull;
-import static org.junit.Assert.assertTrue;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import spoon.Launcher;
+import spoon.SpoonModelBuilder;
+import spoon.reflect.cu.CompilationUnit;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
 
 /**
  * Tests {@link CompilationUnit#getBinaryFiles()}.
  */
 public class GetBinaryFilesTest {
 
-	@Rule
-	public TemporaryFolder tmpFolder = new TemporaryFolder();
+	@TempDir
+	Path tmpFolder;
+
+	@BeforeEach
```

Comment:
This looks strange?

Reply:
Sadly, JUnit 5 has no longer the `@Rule` annotations but extensions, see https://junit.org/junit5/docs/current/user-guide/#migrating-from-junit4-rule-support. The replacement for the old temporary folder concept is https://junit.org/junit5/docs/5.4.2/api/org/junit/jupiter/api/io/TempDir.html, https://www.baeldung.com/junit-5-temporary-directory.

Reply:
Maybe I'm dense here, but where does `@BeforeEach` come into the picture?

I actuall quite like the new `@TempDir` construct, allows you to remove the field (ugh, fields!) and do something like this:

```java
@Test
void testSomething(@TempDir workdir) {
    // use injected temporary directory
}
```

Reply:
Oh the `@BeforeEach` should be removed you are right

Reply:
> I actuall quite like the new @TempDir construct, allows you to remove the field (ugh, fields!) and do something like this:

Yes, if you use this style it's nice, but then the whole codebase should do it this way and not mix of different styles

Reply:
> Yes, if you use this style it's nice, but then the whole codebase should do it this way and not mix of different styles

Good practices have to start somewhere :). That's kind of the thing with legacy, good practices have to be adopted gradually. And often, they are never _fully_ adopted, but we can be watchful for future PRs.

If someone were to submit a PR with a JUnit5 test that had a field like this, I'd request a change. In this case, it's already there, so it's ok.